### PR TITLE
Add changelog, comments and readme for Microsoft.Extensions.Azu…

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Common/swagger/Generator/package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "description": "AutoRest extension to generate C# for Azure Track 2 client libraries",
   "main": "dist/index.js",
+  "private": true,
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Common/swagger/Generator/package.json]: This looks like an accidental inclusion

File: sdk/core/Microsoft.Extensions.Azure/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Release History
+
+## 1.0.0
+
+- Updated Azure.Identity dependency version
```
Comment (tg-msft) [inline sdk/core/Microsoft.Extensions.Azure/CHANGELOG.md]: Didn't we do a bunch of renames?

File: sdk/core/Microsoft.Extensions.Azure/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Azure client integration for ASP.NET Core
+
+Microsoft.Extensions.Azure.Core provides shared primitives to integrate Azure clients with ASP.NET Core dependency injection and configuration systems.
```
Comment (tg-msft) [inline sdk/core/Microsoft.Extensions.Azure/README.md]: Is there a good ASP.NET configuration doc on MSDN we could link folks too if they want to discover more?

File: sdk/core/Microsoft.Extensions.Azure/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+Install the ASP.NET Core integration library using [NuGet][nuget]:
+
+```
+dotnet add package Microsoft.Extensions.Azure
+```
+
+## Usage Scenarios and Samples
+
+Use add `AddAzureClients` call to you `ConfigureServices` method. You can use provided builder to register client instances with dependency injection container.
```
Comment (tg-msft) [inline sdk/core/Microsoft.Extensions.Azure/README.md]: Nit - "Make a call to `AddAzureClients` in your app's `ConfigureServices` method.  You can use the provided builder to register client instances with your dependency injection container."

File: sdk/core/Microsoft.Extensions.Azure/README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+```C# Snippet:ConfigureServices
+public void ConfigureServices(IServiceCollection services)
+{
+    // Registering policy to use in ConfigureDefaults later
+    services.AddSingleton<DependencyInjectionEnabledPolicy>();
+
+    services.AddAzureClients(builder => {
+
+        builder.AddSecretClient(Configuration.GetSection("KeyVault"))
```
Comment (tg-msft) [inline sdk/core/Microsoft.Extensions.Azure/README.md]: Nit - Add a comment like "// Configure a KeyVault SecretClient using ..." to help add context.  If I wasn't familiar with KeyVault, I might think AddSecretClient was doing some random DI thing I had never heard of before.

File: sdk/core/Microsoft.Extensions.Azure/README.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            .WithName("Default")
+            .WithCredential(new DefaultAzureCredential())
+            .ConfigureOptions(options => options.Retry.MaxRetries = 10);
+
+        builder.AddSecretClient(new Uri("http://my.keyvault.com"));
+
+        builder.UseCredential(new DefaultAzureCredential());
```
Comment (tg-msft) [inline sdk/core/Microsoft.Extensions.Azure/README.md]: Add a brief comment explaining what this line does (and maybe why'd you use it relative to some of the other default mechanisms)

File: sdk/core/Microsoft.Extensions.Azure/src/AzureClientBuilderExtensions.cs [hunk 7]
```diff
@@ -10,43 +10,103 @@
...
+        /// <returns>The client builder instance.</returns>
         public static IAzureClientBuilder<TClient, TOptions> ConfigureOptions<TClient, TOptions>(this IAzureClientBuilder<TClient, TOptions> builder, Action<TOptions, IServiceProvider> configureOptions) where TOptions : class
         {
             var impl = builder.ToBuilder();
             impl.ServiceCollection.AddSingleton<IConfigureOptions<TOptions>>(provider => new ConfigureClientOptions<TClient,TOptions>(provider, impl.Registration, configureOptions));;
             return builder;
         }
 
+        /// <summary>
+        /// Sets the service version to use for this client registration. Latest by default.
```
Comment (tg-msft) [inline sdk/core/Microsoft.Extensions.Azure/src/AzureClientBuilderExtensions.cs]: "Latest by default" is a little tricky.  It's `Latest` if you don't call `WithVersion` but `WithVersion(default(TOptions.ServiceVersion))` would not give you Latest for Storage.

File: sdk/core/Microsoft.Extensions.Azure/src/AzureClientFactoryBuilder.cs [hunk 8]
```diff
@@ -91,12 +108,21 @@ public AzureClientFactoryBuilder ConfigureDefaults(IConfiguration configuration)
...
+        /// <returns>This instance.</returns>
         public AzureClientFactoryBuilder UseCredential(TokenCredential tokenCredential)
         {
             return UseCredential(_ => tokenCredential);
         }
 
+        /// <summary>
+        /// Sets the credential to use by default for all clients.
+        /// </summary>
+        /// <param name="tokenCredentialFactory">The credential to use.</param>
```
Comment (tg-msft) [inline sdk/core/Microsoft.Extensions.Azure/src/AzureClientFactoryBuilder.cs]: Nit - `The credential factory to use.`

**Review by tg-msft — APPROVED** [review]
Looks good

File: sdk/core/Microsoft.Extensions.Azure/CHANGELOG.md (hunk 2 above)
Comment (pakrym) [reply sdk/core/Microsoft.Extensions.Azure/CHANGELOG.md]: Not in Microsoft.Extensions.Azure

File: sdk/core/Microsoft.Extensions.Azure/README.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+```C# Snippet:ConfigureServices
+public void ConfigureServices(IServiceCollection services)
+{
+    // Registering policy to use in ConfigureDefaults later
+    services.AddSingleton<DependencyInjectionEnabledPolicy>();
+
+    services.AddAzureClients(builder => {
+        // Register blob service client and initialize it using the KeyVault section of configuration
+        builder.AddSecretClient(Configuration.GetSection("KeyVault"))
```
Comment (heaths) [inline sdk/core/Microsoft.Extensions.Azure/README.md]: Would it make sense to have an example of the .config file that corresponds to this?

**Review by heaths — APPROVED** [review]

File: sdk/core/Microsoft.Extensions.Azure/README.md (hunk 9 above)
Comment (pakrym) [reply sdk/core/Microsoft.Extensions.Azure/README.md]: Yes, adding.

**Comment by pakrym** [comment]
Unrelated failure: https://github.com/Azure/azure-sdk-for-net/issues/7634

**Pull request merged**